### PR TITLE
Issue #28588: Reschedule dates in recurring xtConnect Batch jobs

### PR DESCRIPTION
**Pull request by andersdd**
New function to determine if a field value is a date type - required for xtConnect changes but posting here as it is a generically useful function.

### CODE REVIEW

**Comment by rkannadevara** [comment]
Unable to perform UAT: Observed an error message saying 'Invalid incident category for viewing incident number 28588 within this profile.' is displayed when issue #28588 is opened in bug tracker.

**Comment by andersdd** [comment]
@rkannadevara I have changed the incident category.  The CI build fail is an unrelated issue that I (someone) needs to investigate.

**Comment by lokeshsh** [comment]
Performed UAT: Observed the below error messages on build_app the pull database
Error Message:
Saved PEGjs parser Javascript to  ../lib/orm/source/xt/javascript/parsemetasql.sql
Saved PEGjs parser Javascript to  ../lib/orm/source/xt/javascript/parseediprofile.sql
No built client file for commercialcore. There is probably no client-side code in the extension.
Applying build to database 28588
Cannot install file  /home/xtuple/src/xtuple/scripts/output/build_28588.sql
See errors in  /home/xtuple/src/xtuple/scripts/output/build_28588.sql.log

################################################################################
# ERROR: psql stderr OUTPUT LAST 20 LINES OF: /home/xtuple/src/xtuple/scripts/output/build_28588.sql.log
################################################################################

 psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:338405: NOTICE:  function _gettargetdocument(pg_catalog.int4,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:338810: NOTICE:  function adjustinvvalue(pg_catalog.int4,pg_catalog.numeric,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:341995: NOTICE:  function public.calcinvoiceamt(pg_catalog.int4,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:345236: NOTICE:  function checkprivilege(text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:349671: NOTICE:  function copyso(pg_catalog.int4,date) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:352640: NOTICE:  function createcheck(pg_catalog.int4,text,pg_catalog.int4,date,pg_catalog.numeric,pg_catalog.int4,pg_catalog.int4,pg_catalog.int4,text,text,pg_catalog.bool,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:353347: NOTICE:  function createdoctype(pg_catalog.int4,text,text,text,text,text,text,text,text,text,text,text,text,text,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:353348: NOTICE:  function createdoctype(pg_catalog.int4,text,text,text,text,text,text,text,text,text,text,text,text,text,text,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:354197: NOTICE:  function createpkgschema(text,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:354537: NOTICE:  function createpriv(text,text,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:360209: NOTICE:  function deletewo(pg_catalog.int4,pg_catalog.bool) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:361100: NOTICE:  function distributevoucherline(pg_catalog.int4,pg_catalog.int4,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:361980: NOTICE:  function enterporeturn(pg_catalog.int4,pg_catalog.numeric,pg_catalog.int4,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:372428: NOTICE:  function getunassignedaccntid() does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:375566: NOTICE:  drop cascades to rule _INSERT on view api.itemcost
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:375647: NOTICE:  function invadjustment(pg_catalog.int4,pg_catalog.numeric,text,text) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:375649: NOTICE:  function invadjustment(pg_catalog.int4,pg_catalog.numeric,text,text,pg_catalog.timestamptz) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:375651: NOTICE:  function invadjustment(pg_catalog.int4,pg_catalog.numeric,text,text,pg_catalog.timestamptz,pg_catalog.numeric) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:375653: NOTICE:  function invadjustment(pg_catalog.int4,pg_catalog.numeric,text,text,pg_catalog.timestamptz,pg_catalog.numeric,pg_catalog.int4) does not exist, skipping
psql:/home/xtuple/src/xtuple/scripts/output/build_28588.sql:375996: ERROR:  function "isdate" already exists with same argument types
 
################################################################################
# END OF psql stderr LOG OUTPUT ################################################
################################################################################


Command failed:  Error: Command failed: 
    at ChildProcess.exithandler (child_process.js:658:15)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:766:16)
    at Process.ChildProcess._handle.onexit (child_process.js:833:5) { [Error: Command failed: ] killed: false, code: 3, signal: null }
{ [Error: Command failed: ] killed: false, code: 3, signal: null }

File: foundation-database/public/functions/isdate.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+create function isdate(s varchar) returns boolean as $$
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/isdate.sql]: `CREATE OR REPLACE`

Comment (andersdd) [inline foundation-database/public/functions/isdate.sql]: @gilmoskowitz  Am I correct in saying the function below is only created in a mobile Db?

```lib/orm/source/xt/functions/is_date.sql
1:create or replace function xt.is_date(text) returns boolean immutable as $$```

Can I move this function into core as opposed to duplicating the same one in `public`?

**Comment by rkannadevara** [comment]
Performed UAT: Observed that 'Processing Error' dialog is displayed when 'Show Completed Actions' or 'Automatically Update' check boxes are enabled in 'xTuple Connect Console' screen and also observed that 'Reschedule', 'Retry ASAP', 'View', 'Edit' and 'Cancel' buttons are in disable mode though the job is selected.

Please suggest.

xtbatch

**Comment by gilmoskowitz** [comment]
This UAT failure is unrelated to the code change here. It's a manifestation of bug 29729.

**Comment by gilmoskowitz** [comment]
+1 when merge conflict is resolved

**Comment by xtuple-ci-user** [comment]
Build finished.

**Pull request merged**